base/number.jl

```diff
@@ -3,6 +3,15 @@
 ## generic operations on numbers ##
 
 isinteger(x::Integer) = true
+iszero(x::Number) = x == 0 # faster than x == zero(x) for complex, bigfloat, etc.
+
+"""
+    iszero(x)
+
+Return `true` if `x == zero(x)`; if `x` is an array, this checks whether
+all of the elements of `x` are zero.
+"""
+iszero(x) = x == zero(x) # fallback method
```

Comment:
> What if a type provides zero but not isequal to compare with Int. Probably there's need of a more general solution?

`==` falls back to `===`, so types that do not provide `==` with `Int` fall back to `=== 0` and yield `false`.

> iszero("cat") seems questionable to me... we don't define isinteger("cat") either.

I would be happy with either fallback definition, appreciating arguments for both strictness and permissiveness in this case.

If some feel strongly about the stricter `iszero` definition while others feel strongly about {`map`/`broadcast` over sparse matrices where the output element types don't provide `zero`}, we can accommodate both desires with a strict `iszero` definition and a `_sparsebc_iszero` which wraps `iszero` but provides the permissive fallback.

So in the interest of advancing this pull request, :thumbsup: for leaving the stricter definition in place. Best!